src/app/plugins/ta.plugin.ts

```diff
@@ -0,0 +1,167 @@
+      await this._handleRegister(message);
+      return;
+    }
+
+    if (subcommand === 'remove') {
+      await this._handleRemove(message);
+      return;
+    }
+
+    message.reply('Invalid sub-command');
+  }
+
+  private _handleAsk(message: IMessage, question: string) {
+    const channelTopic = (<TextChannel>message.channel).topic || '';
+    const hasTA: Boolean = channelTopic.indexOf('TA: ') !== -1;
+
+    if (!hasTA) {
+      message.reply('There are no TAs registered in this channel');
+      return;
+    }
+
+    const TA_tags = this._parseTags(channelTopic.split('| TA: ')[1]);
+    let mentions = '';
+    this.container.guildService
+      .get()
+      .members.filter((member) => TA_tags.some((TA: string) => member.user.tag === TA))
+      .forEach((TA: GuildMember) => {
+        mentions += `${TA.user} `;
+      });
+
+    message.channel.send(`${message.author} asks: \n> ${question}\n${mentions}`);
+  }
+
+  private async _handleRegister(message: IMessage) {
+    const channel: GuildChannel = message.channel as GuildChannel;
+    const channelTopic = (channel as TextChannel).topic || '';
+
+    const hasTA: Boolean = channelTopic.indexOf('TA: ') !== -1;
+    if (!hasTA) {
+      await channel.setTopic(`${channelTopic} | TA: ${message.author.tag}`);
+      return;
+    }
+
+    let existingTAs;
+    if (channelTopic === '') {
+      existingTAs = channelTopic.slice('| TA: '.length);
+    } else {
+      [, existingTAs] = channelTopic.split('| TA: ');
+    }
+
+    if (this._parseTags(existingTAs).includes(message.author.tag)) {
+      message.reply(`You are already registered as a TA for ${channel.name}.`);
+      return;
+    }
+
+    await channel
+      .setTopic(`${channelTopic} ${message.author.tag}`)
+      .then((newChan) => message.reply(`Successfully registered as TA in ${newChan.name}`));
+  }
+
+  private async _handleRemove(message: IMessage) {
+    const channel: GuildChannel = message.channel as GuildChannel;
+    const channelTopic = (channel as TextChannel).topic || '';
+
+    const hasTA: Boolean = channelTopic.indexOf('TA: ') !== -1;
+    if (!hasTA) {
+      message.reply(`You are not a TA in ${channel.name}`);
+      return;
+    }
+
+    let originalTopic, existingTAs;
+    if (channelTopic.indexOf('| TA: ') === 0) {
+      originalTopic = '';
+      existingTAs = channelTopic.slice('| TA: '.length);
+    } else {
+      [originalTopic, existingTAs] = channelTopic.split('| TA: ');
+    }
+
+    const newTAs = this._parseTags(existingTAs).filter((e) => e !== message.author.tag);
+    if (newTAs.length === 0) {
+      await channel.setTopic(originalTopic);
+      return;
+    }
+
+    await channel
+      .setTopic(`${originalTopic} | TA: ${newTAs.join(' ')}`)
+      .then((newChan) => message.reply(`Successfully unregistered as TA in ${newChan.name}`));
+  }
+
+  private _parseTags(list: string): string[] {
+    let temp = list;
+    const tags: string[] = [];
+    while (temp.length > 0) {
```

Comment:
could have just used a comma as delimiter and this wouldve been a lot simpler to implement

Reply:
Should I leave as is, or change it to that now that I've done it the hard way  😆 
I don't mind, for the sake of better code

Reply:
can't comma appear in a name?

just checked: you can